File: P1 2018.2/Questao(2)/java/Q2_2018_2.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Q2_2018_2 {
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        BancoDeDados BD = new BancoDeDados();
        BD.incializarParaTeste();

        String nomeProcurado = sc.nextLine();
        List<String> funcs = BD.getFunc();

        List<String[]> matrizFunc = new ArrayList<>();
        int somaIdades = 0;
        for (int i = 0; i < funcs.size(); i++) {
            matrizFunc.add(funcs.get(i).split("#"));
            somaIdades += Integer.parseInt(matrizFunc.get(i)[1]);

            if (nomeProcurado.equalsIgnoreCase(matrizFunc.get(i)[0])) {

                System.out.print(matrizFunc.get(i)[0]
                        + " - idade: "
                        + matrizFunc.get(i)[1]);
            }
        }

        System.out.println(" - media da idade de todos funcionarios - " + somaIdades / funcs.size());
        sc.close();
    }
}

File: P1 2018.2/Questao(2)/python/Q2_2018_2.py
from BancoDeDados import BancoDeDados
from Funcionario import Funcionario


def main():
    func_nome = input("Insira o Nome do Funcionario: ")

    func_lista_strings = BancoDeDados.get_func()
    func_lista = []

    for s in func_lista_strings:
        func_lista_aux = s.split("#", 2)
        func_aux = Funcionario(nome=func_lista_aux[0], idade=int(func_lista_aux[1]))
        func_lista.append(func_aux)

    media_idades = 0
    for f in func_lista:
        media_idades += f.idade / len(func_lista)

    index = func_lista.index(func_nome)
    if index >= 0:
        print("{} - idade: {} - Media de todos os funcionarios - {:.2f}".format(func_lista[index].nome,
                                                                                func_lista[index].idade, media_idades))


if __name__ == "__main__":
    main()
